### PR TITLE
Add a setting to control whether to enable async jdwp

### PR DESCRIPTION
**Pull request by testforstephen**
This PR relates to https://github.com/microsoft/java-debug/pull/436

### CODE REVIEW

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -737,6 +737,11 @@ async function updateDebugSettings(event?: vscode.ConfigurationChangeEvent) {
             const exceptionFilters = {
                 skipClasses: await substituteFilterVariables(debugSettingsRoot.settings.exceptionBreakpoint.skipClasses),
             };
+
+            let asyncJDWP: string = debugSettingsRoot.settings.jdwp.async;
+            if (asyncJDWP === "auto" && vscode?.env?.appName === "Visual Studio Code - Insiders") {
```
Comment (jdneo) [inline src/configurationProvider.ts:742]: `vscode.env?.appName` I think the `?` after `vscode` is unnecessary?

Comment (testforstephen) [reply src/configurationProvider.ts]: nice.

**Review by jdneo — APPROVED** [review]

**Comment by Eskibear** [comment]
Will this setting appear on settings page for vscode stable editions? If so, shall we explicitly note "for Insiders only" also in this setting's description instead of "experimental" to avoid confusion? Because for stable vscode, I also see some native "experimental" settings that works.

**Comment by testforstephen** [comment]
> Will this setting appear on settings page for vscode stable editions? If so, shall we explicitly note "for Insiders only" also in this setting's description instead of "experimental" to avoid confusion? Because for stable vscode, I also see some native "experimental" settings that works.

This is not for Insiders only, just enable it on Insiders by default. For stable, you still can manually opt in.

**Comment by Eskibear** [comment]
> This is not for Insiders only, just enable it on Insiders by default. For stable, you still can manually opt in.

Sorry for misunderstanding. Then it looks good to me.

**Pull request merged**